### PR TITLE
Php8 multiple signatures

### PR DESCRIPTION
**Pull request by MaXal**

### CODE REVIEW

File: gd/gd.php [hunk 1]
```diff
@@ -223,19 +224,37 @@ function imagecolorallocate ($image, $red, $green, $blue) {}
...
  * </p>
- * @return resource|GdImage|false An image resource will be returned on success. <b>FALSE</b> is returned if
+ * @return GdImage|false An image resource will be returned on success. <b>FALSE</b> is returned if
  * the image type is unsupported, the data is not in a recognised format,
  * or the image is corrupt and cannot be loaded.
  */
 #[Pure]
 function imagecreatefromstring ($image) {}
 
+#[PhpStormStubsElementAvailable('5.3','7.4')]
```
Comment (wbars) [inline gd/gd.php:241]: As the matter of code style I'd group multiple attributes together

File: gd/gd.php [hunk 2]
```diff
@@ -505,11 +525,29 @@ function imagecopyresized ($dst_image, $src_image, $dst_x, $dst_y, $src_x, $src_
  * @param int $height <p>
  * The image height.
  * </p>
- * @return resource|GdImage|false an image resource identifier on success, false on errors.
+ * @return GdImage|false an image resource identifier on success, false on errors.
  */
 #[Pure]
 function imagecreate ($width, $height) {}
 
+#[PhpStormStubsElementAvailable('5.3','7.4')]
```
Comment (wbars) [inline gd/gd.php:533]: Quotes are inconsistent :)

File: openssl/openssl.php [hunk 3]
```diff
@@ -721,6 +890,26 @@ function openssl_verify($data, $signature, $public_key, $algorithm = OPENSSL_ALG
...
+ * @return int|false the length of the sealed data on success, or false on error.
+ * If successful the sealed data is returned in
+ * <i>sealed_data</i>, and the envelope keys in
+ * <i>env_keys</i>.
+ */
+function openssl_seal($data, &$sealed_data, array &$encrypted_keys, array $public_key, $cipher_algo, &$iv = '')
+{
+}
+
+#[PhpStormStubsElementAvailable('5.3','7.4')]
```
Comment (wbars) [inline openssl/openssl.php:912]: Generally such samples can be replaced with `#[PhpStormStubsElementAvailable(to: '7.4')]`

**Comment by wbars** [comment]
💪

File: gd/gd.php [hunk 4]
```diff
@@ -496,6 +515,7 @@ function imagecopymergegray ($dst_im, $src_im, $dst_x, $dst_y, $src_x, $src_y, $
  */
 function imagecopyresized ($dst_image, $src_image, $dst_x, $dst_y, $src_x, $src_y, $dst_w, $dst_h, $src_w, $src_h) {}
 
+#[PhpStormStubsElementAvailable("8.0")]
```
Comment (wbars) [inline gd/gd.php:518]: Single quotes?

File: PDO/PDO.php [hunk 5]
```diff
@@ -1003,6 +1005,30 @@ public function exec ($statement) {}
...
+	 * </p>
+	 * @param mixed $fetch_mode_args <p>
+	 * Arguments of custom class constructor when the <i>mode</i>
+	 * parameter is set to <b>PDO::FETCH_CLASS</b>.
+	 * </p>
+	 * @return PDOStatement|false <b>PDO::query</b> returns a PDOStatement object, or <b>FALSE</b>
+	 * on failure.
+	 * @see PDOStatement::setFetchMode For a full description of the second and following parameters.
+	 */
+	public function query ($statement, $mode = PDO::ATTR_DEFAULT_FETCH_MODE, ...$fetch_mode_args) {}
```
Comment (isfedorov) [inline PDO/PDO.php:1030]: May be it makes sense to add property and return types (including union types) directly to signatures for entities marked with `#[PhpStormStubsElementAvailable('8.0')]` ? Thus we will strict types and improve type inference.

**Comment by isfedorov** [comment]
Should we somehow explicitly notify authors of 3rd party tools about such changes to allow them update tools to support duplicated items in stubs?

**Comment by wbars** [comment]
> Should we somehow explicitly notify authors of 3rd party tools about such changes to allow them update tools to support duplicated items in stubs?

Duplicates existed in stubs before this change, so nothing generally should change

File: gd/gd.php (hunk 1 above)
Comment (MaXal) [reply gd/gd.php:241]: I thought about it but decided to insert the annotation before the PHPDoc for better readability but this is discussable I guess and depends on the navigation patterns:) If GoTo is used, it's better to have attributes together if scrolling - better to have a new attribute above.

**Review by wbars — APPROVED** [review]

**Pull request merged**